pkg/scheduler/monitoring/metrics.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2024 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package monitoring
+
+import (
+	"context"
+	"time"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+
+	"github.com/dapr/dapr/pkg/diagnostics/utils"
+	schedulerv1pb "github.com/dapr/dapr/pkg/proto/scheduler/v1"
+)
+
+var (
+	sidecarsConnectedTotal = stats.Int64(
+		"scheduler/sidecars_connected_total",
+		"The total number of dapr sidecars connected to the scheduler service.",
+		stats.UnitDimensionless)
+	jobsScheduledTotal = stats.Int64(
+		"scheduler/jobs_created_total",
+		"The total number of jobs scheduled.",
+		stats.UnitDimensionless)
+	jobsTriggeredTotal = stats.Int64(
+		"scheduler/jobs_triggered_total",
+		"The total number of successfully triggered jobs.",
+		stats.UnitDimensionless)
+	triggerLatency = stats.Float64(
+		"scheduler/trigger_latency",
+		"The total time it takes to trigger a job from the scheduler service.",
+		stats.UnitMilliseconds)
+)
+
+// RecordSidecarsConnectedCount records the number of dapr sidecars connected to the scheduler service
+func RecordSidecarsConnectedCount() {
+	stats.RecordWithTags(context.Background(), utils.WithTags(sidecarsConnectedTotal.Name()), sidecarsConnectedTotal.M(1))
+}
```

Comment:
```suggestion
func RecordSidecarsConnectedCount(cnt int64) {
	stats.RecordWithTags(context.Background(), utils.WithTags(sidecarsConnectedTotal.Name()), sidecarsConnectedTotal.M(cnt))
}
```
Instead of always setting the value to one and checking how many times it was called later (`utils.NewMeasureView(sidecarsConnectedTotal, []tag.Key{}, view.Count())`), we'll record the actual value we're observing and check the last value (`utils.NewMeasureView(sidecarsConnectedTotal, []tag.Key{}, view.LastValue())`).